webapp/components/login/login_controller.jsx

```diff
@@ -60,9 +60,21 @@ export default class LoginController extends React.Component {
             GlobalActions.redirectUserToDefaultTeam();
         }
 
+        if (this.props.location.query.extra === Constants.SIGNIN_VERIFIED && this.props.location.query.email) {
+            this.refs.password.focus();
+        }
+
         AsyncClient.checkVersion();
     }
 
+    componentWillMount() {
```

Comment:
@crspeller If I move the code to the constructor I receive this error in the console:

```
Warning: setState(...): Can only update a mounted or mounting component. This usually means you called setState() on an unmounted component. This is a no-op. Please check the code for the LoginController component.
```

and if I change for `this.state.loginId = this.props.location.query.email;` I got a lint issue.

What is the best approach ?

thanks